// @aws/aws-codeservices-plugin-for-backstage/dist/esm/AboutField-2bd564e7.esm.js
import { makeStyles, Typography, Grid } from '@material-ui/core';
import React from 'react';

const useStyles = makeStyles((theme) => ({
  links: {
    margin: theme.spacing(2, 0),
    display: "grid",
    gridAutoFlow: "column",
    gridAutoColumns: "min-content",
    gridGap: theme.spacing(3)
  },
  label: {
    color: theme.palette.text.secondary,
    textTransform: "uppercase",
    fontSize: "10px",
    fontWeight: "bold",
    letterSpacing: 0.5,
    overflow: "hidden",
    whiteSpace: "nowrap"
  },
  value: {
    fontWeight: "bold",
    overflow: "hidden",
    lineHeight: "24px",
    wordBreak: "break-word"
  },
  description: {
    wordBreak: "break-word"
  }
}));
const AboutField = ({
  label,
  value,
  gridSizes,
  children
}) => {
  const classes = useStyles();
  const content = React.Children.count(children) ? children : /* @__PURE__ */ React.createElement(Typography, { variant: "body2", className: classes.description }, value || `unknown`);
  return /* @__PURE__ */ React.createElement(Grid, { item: true, ...gridSizes }, /* @__PURE__ */ React.createElement(Typography, { variant: "subtitle2", className: classes.label }, label), content);
};

export { AboutField as A };
