// @aws-sdk/client-codedeploy/dist-cjs/CodeDeploy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CodeDeploy = void 0;
const smithy_client_1 = require("@smithy/smithy-client");
const CodeDeployClient_1 = require("./CodeDeployClient");
const AddTagsToOnPremisesInstancesCommand_1 = require("./commands/AddTagsToOnPremisesInstancesCommand");
const BatchGetApplicationRevisionsCommand_1 = require("./commands/BatchGetApplicationRevisionsCommand");
const BatchGetApplicationsCommand_1 = require("./commands/BatchGetApplicationsCommand");
const BatchGetDeploymentGroupsCommand_1 = require("./commands/BatchGetDeploymentGroupsCommand");
const BatchGetDeploymentInstancesCommand_1 = require("./commands/BatchGetDeploymentInstancesCommand");
const BatchGetDeploymentsCommand_1 = require("./commands/BatchGetDeploymentsCommand");
const BatchGetDeploymentTargetsCommand_1 = require("./commands/BatchGetDeploymentTargetsCommand");
const BatchGetOnPremisesInstancesCommand_1 = require("./commands/BatchGetOnPremisesInstancesCommand");
const ContinueDeploymentCommand_1 = require("./commands/ContinueDeploymentCommand");
const CreateApplicationCommand_1 = require("./commands/CreateApplicationCommand");
const CreateDeploymentCommand_1 = require("./commands/CreateDeploymentCommand");
const CreateDeploymentConfigCommand_1 = require("./commands/CreateDeploymentConfigCommand");
const CreateDeploymentGroupCommand_1 = require("./commands/CreateDeploymentGroupCommand");
const DeleteApplicationCommand_1 = require("./commands/DeleteApplicationCommand");
const DeleteDeploymentConfigCommand_1 = require("./commands/DeleteDeploymentConfigCommand");
const DeleteDeploymentGroupCommand_1 = require("./commands/DeleteDeploymentGroupCommand");
const DeleteGitHubAccountTokenCommand_1 = require("./commands/DeleteGitHubAccountTokenCommand");
const DeleteResourcesByExternalIdCommand_1 = require("./commands/DeleteResourcesByExternalIdCommand");
const DeregisterOnPremisesInstanceCommand_1 = require("./commands/DeregisterOnPremisesInstanceCommand");
const GetApplicationCommand_1 = require("./commands/GetApplicationCommand");
const GetApplicationRevisionCommand_1 = require("./commands/GetApplicationRevisionCommand");
const GetDeploymentCommand_1 = require("./commands/GetDeploymentCommand");
const GetDeploymentConfigCommand_1 = require("./commands/GetDeploymentConfigCommand");
const GetDeploymentGroupCommand_1 = require("./commands/GetDeploymentGroupCommand");
const GetDeploymentInstanceCommand_1 = require("./commands/GetDeploymentInstanceCommand");
const GetDeploymentTargetCommand_1 = require("./commands/GetDeploymentTargetCommand");
const GetOnPremisesInstanceCommand_1 = require("./commands/GetOnPremisesInstanceCommand");
const ListApplicationRevisionsCommand_1 = require("./commands/ListApplicationRevisionsCommand");
const ListApplicationsCommand_1 = require("./commands/ListApplicationsCommand");
const ListDeploymentConfigsCommand_1 = require("./commands/ListDeploymentConfigsCommand");
const ListDeploymentGroupsCommand_1 = require("./commands/ListDeploymentGroupsCommand");
const ListDeploymentInstancesCommand_1 = require("./commands/ListDeploymentInstancesCommand");
const ListDeploymentsCommand_1 = require("./commands/ListDeploymentsCommand");
const ListDeploymentTargetsCommand_1 = require("./commands/ListDeploymentTargetsCommand");
const ListGitHubAccountTokenNamesCommand_1 = require("./commands/ListGitHubAccountTokenNamesCommand");
const ListOnPremisesInstancesCommand_1 = require("./commands/ListOnPremisesInstancesCommand");
const ListTagsForResourceCommand_1 = require("./commands/ListTagsForResourceCommand");
const PutLifecycleEventHookExecutionStatusCommand_1 = require("./commands/PutLifecycleEventHookExecutionStatusCommand");
const RegisterApplicationRevisionCommand_1 = require("./commands/RegisterApplicationRevisionCommand");
const RegisterOnPremisesInstanceCommand_1 = require("./commands/RegisterOnPremisesInstanceCommand");
const RemoveTagsFromOnPremisesInstancesCommand_1 = require("./commands/RemoveTagsFromOnPremisesInstancesCommand");
const SkipWaitTimeForInstanceTerminationCommand_1 = require("./commands/SkipWaitTimeForInstanceTerminationCommand");
const StopDeploymentCommand_1 = require("./commands/StopDeploymentCommand");
const TagResourceCommand_1 = require("./commands/TagResourceCommand");
const UntagResourceCommand_1 = require("./commands/UntagResourceCommand");
const UpdateApplicationCommand_1 = require("./commands/UpdateApplicationCommand");
const UpdateDeploymentGroupCommand_1 = require("./commands/UpdateDeploymentGroupCommand");
const commands = {
    AddTagsToOnPremisesInstancesCommand: AddTagsToOnPremisesInstancesCommand_1.AddTagsToOnPremisesInstancesCommand,
    BatchGetApplicationRevisionsCommand: BatchGetApplicationRevisionsCommand_1.BatchGetApplicationRevisionsCommand,
    BatchGetApplicationsCommand: BatchGetApplicationsCommand_1.BatchGetApplicationsCommand,
    BatchGetDeploymentGroupsCommand: BatchGetDeploymentGroupsCommand_1.BatchGetDeploymentGroupsCommand,
    BatchGetDeploymentInstancesCommand: BatchGetDeploymentInstancesCommand_1.BatchGetDeploymentInstancesCommand,
    BatchGetDeploymentsCommand: BatchGetDeploymentsCommand_1.BatchGetDeploymentsCommand,
    BatchGetDeploymentTargetsCommand: BatchGetDeploymentTargetsCommand_1.BatchGetDeploymentTargetsCommand,
    BatchGetOnPremisesInstancesCommand: BatchGetOnPremisesInstancesCommand_1.BatchGetOnPremisesInstancesCommand,
    ContinueDeploymentCommand: ContinueDeploymentCommand_1.ContinueDeploymentCommand,
    CreateApplicationCommand: CreateApplicationCommand_1.CreateApplicationCommand,
    CreateDeploymentCommand: CreateDeploymentCommand_1.CreateDeploymentCommand,
    CreateDeploymentConfigCommand: CreateDeploymentConfigCommand_1.CreateDeploymentConfigCommand,
    CreateDeploymentGroupCommand: CreateDeploymentGroupCommand_1.CreateDeploymentGroupCommand,
    DeleteApplicationCommand: DeleteApplicationCommand_1.DeleteApplicationCommand,
    DeleteDeploymentConfigCommand: DeleteDeploymentConfigCommand_1.DeleteDeploymentConfigCommand,
    DeleteDeploymentGroupCommand: DeleteDeploymentGroupCommand_1.DeleteDeploymentGroupCommand,
    DeleteGitHubAccountTokenCommand: DeleteGitHubAccountTokenCommand_1.DeleteGitHubAccountTokenCommand,
    DeleteResourcesByExternalIdCommand: DeleteResourcesByExternalIdCommand_1.DeleteResourcesByExternalIdCommand,
    DeregisterOnPremisesInstanceCommand: DeregisterOnPremisesInstanceCommand_1.DeregisterOnPremisesInstanceCommand,
    GetApplicationCommand: GetApplicationCommand_1.GetApplicationCommand,
    GetApplicationRevisionCommand: GetApplicationRevisionCommand_1.GetApplicationRevisionCommand,
    GetDeploymentCommand: GetDeploymentCommand_1.GetDeploymentCommand,
    GetDeploymentConfigCommand: GetDeploymentConfigCommand_1.GetDeploymentConfigCommand,
    GetDeploymentGroupCommand: GetDeploymentGroupCommand_1.GetDeploymentGroupCommand,
    GetDeploymentInstanceCommand: GetDeploymentInstanceCommand_1.GetDeploymentInstanceCommand,
    GetDeploymentTargetCommand: GetDeploymentTargetCommand_1.GetDeploymentTargetCommand,
    GetOnPremisesInstanceCommand: GetOnPremisesInstanceCommand_1.GetOnPremisesInstanceCommand,
    ListApplicationRevisionsCommand: ListApplicationRevisionsCommand_1.ListApplicationRevisionsCommand,
    ListApplicationsCommand: ListApplicationsCommand_1.ListApplicationsCommand,
    ListDeploymentConfigsCommand: ListDeploymentConfigsCommand_1.ListDeploymentConfigsCommand,
    ListDeploymentGroupsCommand: ListDeploymentGroupsCommand_1.ListDeploymentGroupsCommand,
    ListDeploymentInstancesCommand: ListDeploymentInstancesCommand_1.ListDeploymentInstancesCommand,
    ListDeploymentsCommand: ListDeploymentsCommand_1.ListDeploymentsCommand,
    ListDeploymentTargetsCommand: ListDeploymentTargetsCommand_1.ListDeploymentTargetsCommand,
    ListGitHubAccountTokenNamesCommand: ListGitHubAccountTokenNamesCommand_1.ListGitHubAccountTokenNamesCommand,
    ListOnPremisesInstancesCommand: ListOnPremisesInstancesCommand_1.ListOnPremisesInstancesCommand,
    ListTagsForResourceCommand: ListTagsForResourceCommand_1.ListTagsForResourceCommand,
    PutLifecycleEventHookExecutionStatusCommand: PutLifecycleEventHookExecutionStatusCommand_1.PutLifecycleEventHookExecutionStatusCommand,
    RegisterApplicationRevisionCommand: RegisterApplicationRevisionCommand_1.RegisterApplicationRevisionCommand,
    RegisterOnPremisesInstanceCommand: RegisterOnPremisesInstanceCommand_1.RegisterOnPremisesInstanceCommand,
    RemoveTagsFromOnPremisesInstancesCommand: RemoveTagsFromOnPremisesInstancesCommand_1.RemoveTagsFromOnPremisesInstancesCommand,
    SkipWaitTimeForInstanceTerminationCommand: SkipWaitTimeForInstanceTerminationCommand_1.SkipWaitTimeForInstanceTerminationCommand,
    StopDeploymentCommand: StopDeploymentCommand_1.StopDeploymentCommand,
    TagResourceCommand: TagResourceCommand_1.TagResourceCommand,
    UntagResourceCommand: UntagResourceCommand_1.UntagResourceCommand,
    UpdateApplicationCommand: UpdateApplicationCommand_1.UpdateApplicationCommand,
    UpdateDeploymentGroupCommand: UpdateDeploymentGroupCommand_1.UpdateDeploymentGroupCommand,
};
class CodeDeploy extends CodeDeployClient_1.CodeDeployClient {
}
exports.CodeDeploy = CodeDeploy;
(0, smithy_client_1.createAggregatedClient)(commands, CodeDeploy);
